packages/vite/src/modes/svelte-scoped/transform.ts

```diff
@@ -0,0 +1,129 @@
+  } = options
+
+  let styles = ''
+
+  const preflights = code.includes('uno:preflights')
+  const safelist = code.includes('uno:safelist')
+
+  if (preflights || safelist) {
+    const { css } = await uno.generate('', { preflights, safelist })
+    styles = css
+  }
+
+  const classes = [...code.matchAll(classesRE)]
+  const classDirectives = [...code.matchAll(classesDirectivesRE)]
+  const classDirectivesShorthand = [...code.matchAll(classDirectivesShorthandRE)]
+
+  if (classes.length || classDirectives.length || classDirectivesShorthand.length) {
+    const originalShortcuts = uno.config.shortcuts
+    const shortcuts: Record<string, string[]> = {}
+    const toGenerate = new Set<string>()
+    const s = new MagicString(code)
+
+    function queueCompiledClass(tokens: string[]) {
+      const hash = hashFn(tokens.join(' '))
+      const className = `${classPrefix}${hash}`
+      shortcuts[className] = tokens
+      toGenerate.add(className)
+      return className
+    }
+
+    async function sortKnownAndUnknownClasses(str: string) {
+      const classArr = str.split(/\s+/)
+      const result = await Promise.all(classArr.filter(Boolean).map(async i => [i, !!await uno.parseToken(i)] as const))
+      const known = result.filter(([, matched]) => matched).map(([i]) => i).sort()
+      if (!known.length)
+        return null
+      const replacements = result.filter(([, matched]) => !matched).map(([i]) => i) // unknown
+      const className = queueCompiledClass(known)
+      return [className, ...replacements].join(' ')
+    }
+
+    for (const match of classes) {
+      let body = expandVariantGroup(match[2].trim())
+
+      const inlineConditionals = [...body.matchAll(classesFromInlineConditionalsRE)]
+      for (const conditional of inlineConditionals) {
+        const replacement = await sortKnownAndUnknownClasses(conditional[1].trim())
+        if (replacement)
+          body = body.replace(conditional[0], `'${replacement}'`)
+      }
+
+      const replacement = await sortKnownAndUnknownClasses(body)
+      if (replacement) {
+        const start = match.index!
+        s.overwrite(start + 7, start + match[0].length - 1, replacement)
+      }
+    }
+
+    for (const match of classDirectives) {
+      const token = match[1]
+      const result = !!await uno.parseToken(token)
+      if (!result)
+        return
+      const className = queueCompiledClass([token])
+      const start = match.index! + 'class:'.length
+      s.overwrite(start, start + match[1].length, className)
+    }
+
+    for (const match of classDirectivesShorthand) {
+      const token = match[1]
+      const result = !!await uno.parseToken(token)
+      if (!result)
+        return
+      const className = queueCompiledClass([token])
+      const start = match.index! + 'class:'.length
+      s.overwrite(start, start + match[1].length, `${className}={${token}}`)
+    }
+
+    if (s.hasChanged())
+      code = s.toString()
+
+    // TODO: return a source map
```

Comment:
Does the transformSFC function need to return a source map? The old one didn't but I think it maybe should be doing that. I can look into it later in a future PR, but would be open to advice on it this time around.

Reply:
Ok, in working on #1719 I decided to figure it out myself so I just updated the transformSvelteSFC function to return sourcemaps when applicable.